course_UvA-DL/contrastive-learning/SimCLR.py

```diff
@@ -0,0 +1,865 @@
+# %% [markdown]
+# Methods for self-supervised learning try to learn as much as possible from the data alone, so it can quickly be finetuned for a specific classification task.
+# The benefit of self-supervised learning is that a large dataset can often easily be obtained.
+# For instance, if we want to train a vision model on semantic segmentation for autonomous driving, we can collect large amounts of data by simply installing a camera in a car, and driving through a city for an hour.
+# In contrast, if we would want to do supervised learning, we would have to manually label all those images before training a model.
+# This is extremely expensive, and would likely take a couple of months to manually label the same amount of data.
+# Further, self-supervised learning can provide an alternative to transfer learning from models pretrained on ImageNet since we could pretrain a model on a specific dataset/situation, e.g. traffic scenarios for autonomous driving.
+#
+# Within the last two years, a lot of new approaches have been proposed for self-supervised learning, in particular for images, that have resulted in great improvements over supervised models when few labels are available.
+# The subfield that we will focus on in this tutorial is contrastive learning.
+# Contrastive learning is motivated by the question mentioned above: how are images different from each other?
+# Specifically, contrastive learning methods train a model to cluster an image and its slightly augmented version in latent space, while the distance to other images should be maximized.
+# A very recent and simple method for this is [SimCLR](https://arxiv.org/abs/2006.10029), which is visualized below (figure credit - [Ting Chen et al. ](https://simclr.github.io/)).
+#
+# <center width="100%"><img src="simclr_contrastive_learning.png" width="500px"></center>
```

Comment:
as MD?

Reply:
I have added the png images in the following format now:
`<center width="100%"> ![simclr contrastive learning](simclr_contrastive_learning.png){width="500px"} </center>`
It compiled correctly locally and seems to pass the documentation test as well.

Reply:
cool!